pkgs/applications/networking/instant-messengers/telegram/telegram-desktop/with-webkit.nix

```diff
@@ -0,0 +1,43 @@
+{
+  stdenv,
+  lib,
+  telegram-desktop,
+  makeBinaryWrapper,
+  wrapGAppsHook3,
+  glib-networking,
+  webkitgtk_4_1,
+}:
+
+if stdenv.hostPlatform.isLinux then
+  stdenv.mkDerivation {
+    pname = "${telegram-desktop.pname}-with-webkit";
+    version = telegram-desktop.version;
+    nativeBuildInputs = [
+      makeBinaryWrapper
+      wrapGAppsHook3
+    ];
+    buildInputs = [
+      glib-networking
+      webkitgtk_4_1
+    ];
+    dontUnpack = true;
+    dontWrapGApps = true;
+    installPhase = ''
+      mkdir -p $out
+      cp -r ${telegram-desktop}/share $out
+      substituteInPlace $out/share/dbus-1/services/* --replace-fail ${telegram-desktop} $out
+    '';
+    postFixup = ''
+      mkdir -p $out/bin
+      makeBinaryWrapper {${telegram-desktop},$out}/bin/${telegram-desktop.meta.mainProgram} \
```

Comment:
It occurred to me: this is double wrapping right? Maybe we should follow the route of having a `telegram-desktop-unwrapped`.

Reply:
Yeah, it is... I'm not sure whether it would be trivial to get Qt wrapper args in a separate derivation?

Reply:
Me neither, we can ignore the aesthetic issue with double wrapping for now.

Reply:
Surprsingly, I actually managed to get it

Reply:
Would this be ok?

Another idea is to copy the unwrapped binary instead of depending on the non-webkit telegram-desktop package.

Reply:
My idea is to rename the original tdesktop package to tdesktop-unwrapped (minus the wrapper part), then create a new tdesktop package for the wrapper, with the argument withWebkitgtk.

Reply:
Hmm... That would mean each override has to provide two packages (unwrapped and wrapped)? And those sitting in by-name will get -unwrapped in their folder names?

Reply:
Not necessarily, as in https://github.com/NickCao/nixpkgs/commit/303fc788531ce4f1c5a84b624456964a88ed991b

Reply:
Oh, ok. Will you submit it? kotato looks broken though as it switches it from Qt 5 to Qt 6 which won't work...

P.S. reading the diff I noticed I forgot to remove kAudioObjectPropertyElementMain -> kAudioObjectPropertyElementMaster replacement which shouldn't be needed with the new sdk...

Reply:
Fixed, see https://github.com/NixOS/nixpkgs/pull/353524